src/writers/file.js

```diff
@@ -0,0 +1,29 @@
+const fs = require('fs')
+const mkdirp = require('mkdirp')
+const path = require('path')
+
+function write (path, content) {
+  try {
+    ensurePath(path)
+    fs.writeFileSync(path, content, 'utf8')
+  } catch (e) {
+    throw new Error(`Failed to write configuration to ${path} because: ${e.message}`)
+  }
+}
+
+function prepare (target) {
+  ensurePath(target)
+}
+
+function ensurePath (fullPath) {
+  const directoryPath = path.extname(fullPath)
```

Comment:
We can replace this with just the call to `mkdirp` since it'll succeed if the path already exists or it'll make it for us.